sheets/1/exercise-3.tex

```diff
@@ -2,27 +2,35 @@
 \excercise{Syntax}
 
 In dieser Aufgabe geht es darum den Syntax von Java zu verstehen.
-In jeder teilaufgabe wir ein bestimmter teil von Java erklärt und du sollst ein Beispiel aus Aufgabe 1 finden. In \textbf{[]} steht immer wie viele verschiedenen Beispiele du finden sollst.
+In jeder teilaufgabe wir ein bestimmter teil von Java erklärt und du sollst ein Beispiel aus Aufgabe 1 finden. In \textbf{[]} steht immer wie viele verschiedenen Beispiele du finden solltest.
 
 \begin{enumerate}
     \item \textbf{[1]} Eine Klasse hat immer am Anfang das Schlüsselwort \lstinline{class}, gefolgt von einem frei wählbaren Namen. Darauf folgen die \{ \}, die den gesamten Inhalt der Klasse umschließen.
           Zum inhalt einer Klasse gehören \textbf{[1]}Attribute und Operationen
 
-          \textbf{[1]} Eine Operationen hat immer einen frei wählbaren Namen, vor dem entweder void oder eine Klasse steht. Eventuell gefolgt von Parametern. Die immer als Typ-Name (Klassen definieren Typen, Also kann man hier einfach den Namen einer Bestehenden Klasse hinschreiben) und dann ein neuer Name für die Operation, geschreiben werden und wenn man mehrere braucht kann man sie mit \lstinline{,} getrennt auflisten.
-          Nach den \lstinline{()} folgen \texttt{ \{\} }, die einen Codeblock umschließen.
+          \textbf{[1]} Eine Operationen hat immer einen frei wählbaren Namen, vor dem ein sogenannter Rückgabetyp steht.
+          Einen Rückgabetypen kann mit einem Klassennamen angeben oder mit \lstinline{void}, was gleich bedeutend mit keinem      Rückgabetyp ist.
+          Eine Operation sollte einen Sichtbarkeits-Modifier haben.
+          Was ein Modifier genau macht wird noch erklährt, deshalb verwenden wir einfach mal immernur \lstinline{public}.
+          Alle Modifier kommen noch vor den Rückgabetypen.
+          Auf der anderen Seiten vom Namen müssen \lstinline{()} stehen.
+          In diesen können Parameter stehen. Damit verlangt die operation nach zusätzliche Informationen, die sie braucht. Ein Parameter besteht immer aus einem Typ gefolgt von einem Namen.
+          Beim Typ verhält es sich genau so wie beim Rückgabetyp vorhin. Nur das hier \lstinline{void} nicht erlaubt ist.
+          Sollte man mehrer Parameter brauchen, kann man die mit einem \lstinline{,} trennen.
+          Nach den \lstinline{()} folgen \texttt{ \{\} }, die einen Codeblock umschließen. Darin Stehen alle Anweisungen, die in den Folge Beispielen genauer erläutert werden.
 
           \begin{lstlisting}[title={\textbf{Klassen Syntax}}]
 class <Name der Klasse>{
-        
-    void <Operation Name>(<Typ Name> <Parameter Name>){
-                
-    }                    
+
+    <Modifier> <Rückgabetyp> <Name>(<Typ> Name,<Typ> <Name>){
```

Comment:
```suggestion
    <Modifier> <Rückgabetyp> <(Operations-)Name>(<Typ> <(Parameter-)Name>,<Typ> <Name>){
```
In (Klammern) wäre optional

Reply:
Entweder Operationsname oder Parametername wäre schon gut, damit man nicht drei mal gleich Name dastehen hat wenn es um ganz unterschiedliche Namen geht.